src/components/app.js

```diff
@@ -338,6 +339,11 @@ const App = () => {
         });
     });
 
+    // Manage user (both manual and auto) disconnection
+    useEffect(() => {
```

Comment:
I would have put this code in app-top-bar.js since it seems strange to me to handle the currentTreeNode redux storage so high in the hierarchy.